dependencies/O2PhysicsCompileFlags.cmake

```diff
@@ -11,7 +11,14 @@
 
 include_guard()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-error -Werror=unused-variable")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-error \
+-Werror=aggressive-loop-optimizations \
+-Werror=deprecated-enum-float-conversion \
+-Werror=maybe-uninitialized \
```

Comment:
This is a GCC-specific option, as well as `-Werror=aggressive-loop-optimizations`, you should add clang's alternatives based on the platform.